.github/workflows/pr-test.yml

```
@@ -29,33 +29,28 @@ jobs:
             gradle-packages-${{ runner.os }}
 
       - name: Unit test
-        run: ./gradlew jacocoTestReport 
-        
-      - name: Copy unit test reports
-        run: ./gradlew copyAllReports --output-dir=/tmp/test-reports
+        run: ./gradlew -no-build-cache cleanTest jacocoTestReport
```

```suggestion
        run: ./gradlew --no-build-cache jacocoTestReport
```

Any reason for the clean?

Alternatively though, what if we add 
```
codecov:
  max_report_age: off
``` 
to our codecov yaml? That should both let us verify this is the issue in PR, but also allow us to continue to take advantage of caching.

---

we can try that. but let's just go with this for now as I want to use this action: https://github.com/burrunan/gradle-cache-action for handling caching. It has good features and OpenTelemetry also uses it: https://github.com/open-telemetry/opentelemetry-java/blob/master/.github/workflows/pr-build.yml

---

and also of we resolve this with codecov for now we have to add that codecov config in every repo.

---

I'm not sure how the two are related - the linked action looks neat, but it optimizes gradle's cache via the GH one (and adds support for other gradle features like remote caches). Disabling caching on this command would break that action too if we don't change the codecov config.

---

Ahh. okay. maybe it was not causing any issue for me because it was not using out cache. 

Makes sense to me then. Let's add it here.